metaci/api/views/robot.py

```diff
@@ -0,0 +1,33 @@
+from rest_framework.renderers import JSONRenderer, BrowsableAPIRenderer
+from rest_framework import viewsets
+from django_filters.rest_framework import DjangoFilterBackend
+from metaci.api.renderers import SimpleCSVRenderer
+from metaci.api.serializers.robot import RobotTestResultSerializer
+from metaci.testresults.models import TestResult
+from metaci.testresults.filters import RobotResultFilter
+from rest_framework.filters import OrderingFilter
+
+
+class RobotTestResultViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    A viewset for viewing robot framework test results
+    """
+
+    serializer_class = RobotTestResultSerializer
+    renderer_classes = [BrowsableAPIRenderer, JSONRenderer, SimpleCSVRenderer]
+    filterset_class = RobotResultFilter
+
+    def get_queryset(self):
+        queryset = TestResult.objects.filter(method__testclass__test_type="Robot")
+
+        # FIXME: I'd like the user to be able to specify a period
+        # of time such as "last two weeks" or "this month"...
+        # I also need to
+
+        # this is probably not the best default, but it's
+        # good enough for now. Eventually I should add django's
+        # ordering filter.
+        sort = self.request.query_params.get("sort", "id").split(",")
+        queryset = queryset.order_by(*sort)
+
+        return queryset
```

Comment:
This may be a good place to use https://docs.djangoproject.com/en/3.0/ref/models/querysets/#prefetch-related so that the related objects needed for filtering and serializing are fetched in a single query rather than one per subobject. I suspect that may be a key source of the slow performance you mentioned. Django Debug Toolbar should be useful to check what queries are actually happening.

Reply:
ah, `prefetch_related` is something I always have to be reminded about. I make this mistake once every couple of years. Thanks!